src/Microsoft.AspNet.Mvc.Core/ModelBindingHelper.cs

```diff
@@ -530,5 +524,189 @@ public static object ConvertValuesToCollectionType<T>(Type modelType, IList<T> v
+            return UnwrapPossibleArrayType(value, type, cultureToUse);
+        }
+
+        private static object UnwrapPossibleArrayType(object value, Type destinationType, CultureInfo culture)
+        {
+            // array conversion results in four cases, as below
+            var valueAsArray = value as Array;
+            if (destinationType.IsArray)
+            {
+                var destinationElementType = destinationType.GetElementType();
+                if (valueAsArray != null)
+                {
+                    // case 1: both destination + source type are arrays, so convert each element
+                    var converted = (IList)Array.CreateInstance(destinationElementType, valueAsArray.Length);
+                    for (var i = 0; i < valueAsArray.Length; i++)
+                    {
+                        converted[i] = ConvertSimpleType(valueAsArray.GetValue(i), destinationElementType, culture);
+                    }
+                    return converted;
+                }
+                else
+                {
+                    // case 2: destination type is array but source is single element, so wrap element in
+                    // array + convert
+                    var element = ConvertSimpleType(value, destinationElementType, culture);
+                    var converted = (IList)Array.CreateInstance(destinationElementType, 1);
+                    converted[0] = element;
+                    return converted;
+                }
+            }
+            else if (valueAsArray != null)
+            {
+                // case 3: destination type is single element but source is array, so extract first element + convert
+                if (valueAsArray.Length > 0)
+                {
+                    value = valueAsArray.GetValue(0);
+                    return ConvertSimpleType(value, destinationType, culture);
+                }
+                else
+                {
+                    // case 3(a): source is empty array, so can't perform conversion
+                    return null;
+                }
+            }
+
+            // case 4: both destination + source type are single elements, so convert
+            return ConvertSimpleType(value, destinationType, culture);
+        }
+
+        private static object ConvertSimpleType(object value, Type destinationType, CultureInfo culture)
+        {
+            if (value == null || value.GetType().IsAssignableFrom(destinationType))
+            {
+                return value;
+            }
+
+            // In case of a Nullable object, we try again with its underlying type.
+            destinationType = UnwrapNullableType(destinationType);
+
+            // if this is a user-input value but the user didn't type anything, return no value
+            var valueAsString = value as string;
+            if (valueAsString != null && string.IsNullOrWhiteSpace(valueAsString))
+            {
+                return null;
```

Comment:
:ok_hand: I was overly-focused on the special case in which `ConvertTo(Type)` activates an instance.